kvbc/src/v4blockchain/v4_blockchain.cpp

```diff
@@ -708,16 +708,15 @@ void KeyValueBlockchain::checkpointInProcess(bool flag, kvbc::BlockId block_id_a
   // shouldn't have any contention or overhead
   const std::lock_guard<std::mutex> lock(map_mutex);
   if (flag) {
-    auto last_reachable_id = block_chain_.getLastReachable();
-    ConcordAssertEQ(last_reachable_id, block_id_at_chkpnt);
     auto new_snap_shot = RecoverySnapshot{&native_client_->rawDB()};
-    chkpnt_snap_shots_[last_reachable_id] = new_snap_shot.get();
+    chkpnt_snap_shots_[block_id_at_chkpnt] = new_snap_shot.get();
```

Comment:
This is incorrect; the newly created snapshot represents the current state of the blockchain i.e. the state of the latest block id, while the change assumes that it represents a block id at the checkpoint, which is a block in the past.